app/views/helpers/injected/obligations/NextUpdatesHelper.scala.html

```diff
@@ -69,27 +70,45 @@
 @getSectionHeadingMessage(obligations.head.obligation)
 }
 
-@bodyContent(collectedObligations: Seq[NextUpdateModelWithIncomeType]) = @{
-    govukTable(Table(
-        rows = {
-            for(obligation <- collectedObligations) yield deadlineRow(obligation)
-        },
-        head = Some(Seq(
-            HeadCell(
-                content = Text(messages("nextUpdates.table.header.type"))
-            ),
-            HeadCell(
-                content = Text(messages("nextUpdates.table.header.source"))
+@makeTable(collectedObligations: Seq[NextUpdateModelWithIncomeType]) = @{
+    collectedObligations match {
+        case validObligations: Seq[NextUpdateModelWithIncomeType] if validObligations.nonEmpty =>
+            govukTable(Table(
+                rows = {
+                    for(obligation <- validObligations) yield deadlineRow(obligation)
+                },
+                head = Some(Seq(
+                    HeadCell(
+                        content = Text(messages("nextUpdates.table.header.type"))
+                    ),
+                    HeadCell(
+                        content = Text(messages("nextUpdates.table.header.source"))
+                    )
+                )),
+                caption = {
+                    if(validObligations.head.obligation.obligationType == "Quarterly")
+                        Some(getTableHeadingMessage(validObligations.head.obligation)) else None
+                },
+                captionClasses = "govuk-table__caption  govuk-!-padding-bottom-2"
             )
-        )),
-        caption = {
-            if(collectedObligations.head.obligation.obligationType == "Quarterly")
-                Some(getTableHeadingMessage(collectedObligations.head.obligation)) else None
-        },
-        captionClasses = "govuk-table__caption  govuk-!-padding-bottom-2",
-        attributes = Map("id"->"next-updates-table")
-    )
-    )
+            )
+        case _ => HtmlFormat.empty
+    }
+
+}
+
+@bodyContent(collectedObligations: Seq[NextUpdateModelWithIncomeType]) = @{
```

Comment:
might worth to move this within service layer so this can be tested for various scenarios;
there is some complexity involved in here and also worth to consider to change view model (add new view model if needed), depending how much time we can spend on this in this sprint